testsuite/MDAnalysisTests/topology/test_pdb.py

```diff
@@ -310,12 +309,30 @@ def test_wrong_elements_warnings():
     column which have been parsed and returns an appropriate warning.
     """
     with pytest.warns(UserWarning, match='Unknown element XX found'):
-        u = mda.Universe(StringIO(PDB_wrong_ele), format='PDB')
+        u = mda.Universe(StringIO(PDB_wrong_ele,), 
+                         force_guess=("types", "masses",), format='PDB')
 
-    expected = np.array(['N', '', 'C', 'O', '', 'Cu', 'Fe', 'Mg'],
+    expected_elements = np.array(['N', '', 'C', 'O', '', 'Cu', 'Fe', 'Mg'],
                         dtype=object)
-    assert_equal(u.atoms.elements, expected)
+    gussed_types = np.array(['N', 'C', 'C', 'O', 'X', 'CU', 'FE', 'MG'])
+    guseed_masses = np.array([14.007, 0.0, 12.011, 15.999,  0.0,
+                               63.546, 55.847, 24.305], dtype=float)
+
+    assert_equal(u.atoms.elements, expected_elements)
+    assert_equal(u.atoms.types, gussed_types)
+    assert_allclose(u.atoms.masses, guseed_masses)
+
+def test_guessed_masses_and_types_values():
+    """Test that guessed masses and types have the expected values for universe
+       constructed from PDB file.
+    """
+    u = mda.Universe(PDB, force_guess=("types", "masses"), format='PDB')
+    gussed_types = np.array(['N', 'H', 'H', 'H', 'C', 'H', 'C', 'H', 'H', 'C'])
+    guseed_masses = [tables.masses[element.upper()] if element != '' else 0.0 
+                     for element in u.atoms.types]
 
+    assert_allclose(u.atoms.masses, guseed_masses)
+    assert_equal(u.atoms.types[:10], gussed_types)
```

Comment:
```suggestion
    gussed_types = np.array(['N', 'H', 'H', 'H', 'C', 'H', 'C', 'H', 'H', 'C'])
    guseed_masses = [14.007, 1.008, 1.008, 1.008, 12.011, 1.008, 12.011, 1.008, 1.008, 12.011]

    assert_allclose(u.atoms.masses[:10], guseed_masses)
    assert_equal(u.atoms.types[:10], gussed_types)
    failed_type_guesses = u.atoms.types == ""
    assert not failed_type_guesses.any()
```

Hmm, I think the PDB should have fairly standard atom names that should allow guessing for all types. IMO it's a stronger test to first compare directly to the expected masses, instead of a table lookup, and secondly to check that types are never empty.